### DISCUSSION TITLE
Downgrade to mariadb 10.3 form 10.5

### DISCUSSION THREAD

**Discussion by calamusid** [Homebrew/discussions#40] [Everyday usage]
Hi,
I have mariadb 10.5, works fine but I need to downgrade to 10.3.
I have run "brew install mariadb@10.3" but the running version still 10.5

I have tried to remove all mariadb package and install only mariadb@10.3 but dosen't work.

googling on doesn't helmp me

thanks
Piero

**Comment by gromgit** [comment]
When you did `brew install mariadb@10.3`, it would've printed the following at the end:
```
==> Caveats
[...]
mariadb@10.3 is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.


To have launchd start mariadb@10.3 now and restart at login:
  brew services start mariadb@10.3
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/mariadb@10.3/bin/mysql.server start
```
The above explains every one of your symptoms and how to address them, so you should read it carefully.

> I have run "brew install mariadb@10.3" but the running version still 10.5

You had both 10.3 and 10.5 installed, but 10.3 is installed _keg-only_, so it will not be accessible the way 10.5 is. The "secret sauce" to starting a 10.3 instance is given in the caveat above, so I won't repeat it here.

However, if you need to run the 10.3 MariaDB _client_, you can do one of the following:

1. Specify its full pathname: `/usr/local/opt/mariadb@10.3/bin/mysql`
1. Add `/usr/local/opt/mariadb@10.3/bin` to your `PATH`
1. `brew link mariadb@10.3`, which symlinks the 10.3 installation into the usual Homebrew directories, so the client can be run as simply `mysql`

> I have tried to remove all mariadb package and install only mariadb@10.3 but dosen't work.

I hope you ran `brew services stop mariadb` first, or stopped the 10.5 server by some other means before uninstalled it, otherwise you may encounter data loss.
